sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -357,11 +374,12 @@ internal static string BuildMetadataString(Metadata metadata)
         /// </summary>
         internal virtual void SetNameFieldsIfNull()
         {
-            if (_fileSystemName == null || _accountName == null)
+            if (_fileSystemName == null || _accountName == null || _path == null)
             {
                 var builder = new DataLakeUriBuilder(Uri);
                 _fileSystemName = builder.FileSystemName;
                 _accountName = builder.AccountName;
+                _path = builder.DirectoryOrFilePath;
```

Comment:
Should there be a Name property here that uses LastDirectoryOrFileName? If so, the Name properties on Directory/File client can be removed.

Reply:
LastDirectoryOrFileName() is used to populate the Name property on DirectoryClient and FileClient, the child classes of PathClient.